test/events_test.go

```diff
@@ -1,18 +1,88 @@
 package test
 
 import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	events "github.com/checkout/checkout-sdk-go/events/abc"
 )
 
 func TestShouldGetEventTypes(t *testing.T) {
-	response, err := PreviousApi().Events.RetrieveAllEventTypes()
+	response, err := PreviousApi().Events.RetrieveAllEventTypes(
+		events.QueryRetrieveAllEventType{Version: "2.0"},
+	)
 	assert.Nil(t, err)
 	assert.NotNil(t, response)
 
-	eventsResponseVersion, err := PreviousApi().Events.RetrieveAllEventTypes(response.EventTypes[0].Version)
+	eventsResponseVersion, err := PreviousApi().Events.RetrieveAllEventTypes(
+		events.QueryRetrieveAllEventType{Version: response.EventTypes[0].Version},
+	)
 	assert.Nil(t, err)
 	assert.NotNil(t, eventsResponseVersion)
 	assert.Equal(t, response.EventTypes[0].EventTypes, eventsResponseVersion.EventTypes[0].EventTypes)
 }
+
+func TestShouldGetAllEventsOneEventAndEventNotification(t *testing.T) {
```

Comment:
We should try to do this in the test case format that we are using as standard. You could create auxiliary functions to retrieve events and all the things you need to setup each particular test.

Reply:
This tests are nested

Reply:
I will update these test